src/main/java/seedu/address/logic/commands/SortStudentCommand.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SORT_BY;
+
+import java.util.Comparator;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Sorts students information lexicographically based on given parameter
+ */
+public class SortStudentCommand extends Command {
+
+    public static final String COMMAND_WORD = "/sort_student";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts students based on specified parameter.\n"
+            + "Parameters: "
+            + PREFIX_SORT_BY + "name " + "or "
+            + PREFIX_SORT_BY + "id " + "or "
+            + PREFIX_SORT_BY + "email\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_SORT_BY + "name";
+
+    public static final String MESSAGE_SORT_STUDENT_SUCCESS = "Students have been successfully sorted";
+    public static final String MESSAGE_INVALID_PARAMETER = "Not a valid parameter for sorting.";
+    public static final String MESSAGE_EMPTY_PARAMETER = "Please provide a parameter.";
+
+    private final String sortBy;
+    public SortStudentCommand(String sortBy) {
+        this.sortBy = sortBy;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        Comparator<Person> personComparator;
+        switch (sortBy) {
```

Comment:
Good handling of input edge cases here by converting to lower case in the parser. You can consider using Enum types instead of string for determining what to sort by to make it more robust.